core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -545,15 +546,39 @@ class Partition(val topic: String,
     laggingReplicas
   }
 
-  def appendRecordsToFutureReplica(records: MemoryRecords) {
-    getReplica(Request.FutureLocalReplicaId).get.log.get.appendAsFollower(records)
+  private def doAppendRecordsToFollower(records: MemoryRecords, isFuture: Boolean): Unit = {
+      if (isFuture)
+        getReplica(Request.FutureLocalReplicaId).get.log.get.appendAsFollower(records)
```

Comment:
We don't necessarily need to fix it here since this was already a problem, but all these calls to `get` are discomforting. I'm wondering if we should have some helpers which raise (perhaps) `ReplicaNotAvailableException` if the option is empty.

Reply:
That's true... I added Partition#getReplicaOrException and used it in `doAppendRecordsToFollower` + unit tests to verify this. I did not refactor any other code.